core/GlobalState.cc

```diff
@@ -2075,22 +2075,22 @@ MethodRef GlobalState::lookupStaticInitForClass(ClassOrModuleRef klass) const {
     return ref;
 }
 
-MethodRef GlobalState::staticInitForFile(Loc loc) {
-    auto nm = freshNameUnique(core::UniqueNameKind::Namer, core::Names::staticInit(), loc.file().id());
+MethodRef GlobalState::staticInitForFile(FileRef file) {
+    auto nm = freshNameUnique(core::UniqueNameKind::Namer, core::Names::staticInit(), file.id());
     auto prevCount = this->methodsUsed();
+    auto loc = core::Loc::none(file);
```

Comment:
What do you think about making `staticInitForFile` still take a `Loc` instead of just a file so that we don't have to use `Loc::none` and have the `???` show up in the symbol table?

It seems like most of the simplifying changes that you want still happen if only `lookupStaticInitForFile` is changed to take a `FileRef`.

Reply:
Yeah, I'm kind of thinking the compiler-related failures are because we don't have a "real" `Loc` for things, so we might just have to still have this weird "for a file, but actually cares about the position in the file".